nilearn/plotting/matrix_plotting.py

```diff
@@ -80,8 +80,12 @@ def plot_matrix(mat, title=None, labels=None, figure=None, axes=None,
         display : instance of matplotlib
             Axes image.
     """
+    # we need a list so an empty one will be cast to False
+    if isinstance(labels, np.ndarray):
+        labels = labels.tolist()
+
     if reorder:
-        if labels is None or labels is False:
+        if not labels:
```

Comment:
I think iit is better to explicitly check if labels is `None` and if not use `len` or a numpy array's `size` if you need to check if it is empty rather than the implict cast to `bool`.

it would also make sense to check that if labels is not None it must be a sequence of the same size as the plotted matrix number of rows

Reply:
The first point might be a matter of style, but [PEP8](https://www.python.org/dev/peps/pep-0008/) recommends exactly that - casting the list to `bool`.

I implemented a check for the length of labels, good idea.

Reply:
If `labels` is a pure Python list/tuple then `not labels` works fine. However if it's a numpy array, then the cast to bool is ambiguous & will raise a warning.
If it's possible this will be a numpy array then `is None` & `is False` should be used.

Reply:
the doc doesn't list `False` or an empty list as valid inputs. let's make the docstring more explicit that it should be `None` for no labels, or a sequence of length number of rows in the matrix, raise an error if the input is not one of these two things, build an `ndarray` from the labels if they are not None. and use explicit `is None` when needed.

Reply:
@kchawla-pi it isn't possible that it's a numpy array, and a test is implemented that checks for the correct handling of an ndarray as input
@jeromedockes fair point, the documentation is ambiguous. The motivating example for this PR was an error that arose when `labels=[]`, I think it makes sense to cover that case, keep the implicit casting to bool, then make the docs more explicit. IMO these are all reasonable user inputs and casting to bool makes the code more readable, even if it is somewhat less explicit

Reply:
this is true. it just feels a bit awkward having to cast the array to list just evaluate if its empty and then back to array, but it does allow handling None, False and an empty sequence uniformly. let's keep it the way you wrote it!